### ISSUE TITLE
Core Image Block breaks if size fields are blank

### ISSUE THREAD

**Issue by DMPR-dev** [WordPress/gutenberg#16942]
**Describe the bug**
If I remove all characters from height and width fields, save the post/page and refresh it, I will get block validation failed error.

**To reproduce**
1. Add core image block to post content
2. Save the post, refresh the page(not sure if this step is needed)
3. Edit core image block options, set width and height fields to be empty.
4. Save the post, refresh the page.
5. See error.

**Expected behavior**
I expected it to be just unsized image.

**Screenshots**
https://i.imgur.com/GuSeHgs.png
https://i.imgur.com/xjST1Yc.png
https://i.imgur.com/kTQxVDO.png

**Desktop (please complete the following information):**
 - OS: MacOS Mojave 10.14.2
 - Browser: Chrome
 - Version 75.0.3770.142 x64
 - Wordpress 5.2.2

I've experienced it with both my custom and builtin themes.

**Comment by tellthemachines** [comment]
Can reproduce this bug, but steps are slightly different now because it is no longer possible to remove the width and height completely from the image. Steps to reproduce are now:

* Delete characters from the width or height box, one by one. The image progressively gets smaller in size.
* When the last character is deleted, the original number pops back into the box, but the image doesn't return to the original size.
* Updating the post and refreshing the page results in block validation error for the image block.
* Converting invalid block to HTML reveals img now has `width="NaN"` attribute.

**Comment by paaljoachim** [comment]
Testing with WordPress 5.7.1. Twenty Twenty One. Gutenberg plugin 10.5.2.

Made a new post
Inserted an Image block. Inserted image.
Removed the Width and Height in the sidebar Image block settings.
Published post. Refreshed the browser.
Noticed this:

![Screen Shot 2021-04-29 at 20 22 18](https://user-images.githubusercontent.com/5323259/116599695-b7100b00-a928-11eb-9e24-493c4b9c3c68.jpg)


JS Console errors:
```
Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(2), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(2), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(1), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt=""/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(2), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(2), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(2), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Expected attributes Array(1), instead saw Array(4).
(anonymous) @ index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13
index.js?ver=32759fecbb9b08f261376daa9b5d86c8:13 Block validation: Block validation failed for `core/image` (Object).

Content generated by `save` function:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt=""/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>

Content retrieved from post body:

<figure class="wp-block-image size-large"><img src="https://test-site.dev.cc/wp-content/uploads/Sun-gradients-1024x1024.jpg" alt="" class="wp-image-283" width="NaN" height="NaN"/><figcaption>This is a very nice sunset with various shades of yellow.</figcaption></figure>
```

**Comment by gwwar** [comment]
I was able to reproduce this too. As @tellthemachines noted, it's important to delete characters from the field. Pressing up and down controls, won't reproduce this.

https://user-images.githubusercontent.com/1270189/116600209-e8171e00-a8dd-11eb-9cbf-6d7578a60e1b.mp4

**Comment by Mamaduka** [comment]
Fixed via #32524.
